File: backend/models/Project.js
const mongoose = require('mongoose');

const ProjectSchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: String,
  url: String, // Live Demo URL
  githubUrl: String,
  role: String, // e.g., "Individual" or "Team Lead"
  tech: [String],
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Project', ProjectSchema);

File: frontend/lib/models/project.dart
class Project {
  final String id;
  final String title;
  final String description;
  final String? url;
  final String? githubUrl;
  final String? role;
  final List<String> tech;

  Project({
    required this.id,
    required this.title,
    required this.description,
    this.url,
    this.githubUrl,
    this.role,
    required this.tech,
  });

  factory Project.fromJson(Map<String, dynamic> json) => Project(
        id: json['_id'] ?? json['id'] ?? '',
        title: json['title'] ?? '',
        description: json['description'] ?? '',
        url: json['url'],
        githubUrl: json['githubUrl'],
        role: json['role'],
        tech: List<String>.from(json['tech'] ?? []),
      );
}
